plugins/wazuh-core/public/services/server-security/ui/components/button.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Wazuh app - Button with Wazuh API permissions and/or roles required to be useful
+ * Copyright (C) 2015-2022 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React from 'react';
+
+import {
+  EuiSwitch,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiLink,
+} from '@elastic/eui';
+
+import {
+  ServerElementPermissionsProps,
+  ServerElementPermissions,
+} from './element';
+
+interface IServerButtonPermissionsProps
+  extends Omit<
+    ServerElementPermissionsProps,
+    'children' | 'additionalPropsFunction'
+  > {
+  buttonType?: 'default' | 'empty' | 'icon' | 'link' | 'switch';
+  rest: any;
+}
+
+export const ServerButtonPermissions = ({
+  buttonType = 'default',
+  permissions,
+  administrator,
+  tooltip,
+  ...rest
+}: IServerButtonPermissionsProps) => {
+  const Button =
+    buttonType === 'empty'
+      ? EuiButtonEmpty
+      : buttonType === 'icon'
+      ? EuiButtonIcon
+      : buttonType === 'link'
+      ? EuiLink
+      : buttonType === 'switch'
+      ? EuiSwitch
+      : EuiButton;
+
+  return (
+    <ServerElementPermissions
+      permissions={permissions}
+      administrator={administrator}
+      tooltip={tooltip}
+      getAdditionalProps={disabled => {
+        const additionalProps = {
+          ...(!['link', 'switch'].includes(buttonType)
+            ? { isDisabled: disabled }
+            : { disabled }),
+          onClick: disabled || !rest.onClick ? undefined : rest.onClick,
+          onChange:
+            !disabled || rest.onChange || buttonType === 'switch'
+              ? rest.onChange
+              : undefined,
+        };
+        if (buttonType == 'switch') delete additionalProps.onClick;
```

Comment:
Use the enum created in the previous comment.

Reply:
Fixed here: https://github.com/wazuh/wazuh-dashboard-plugins/pull/7026/commits/bb3ca50c7a90c743c1ac633c508103e6d3eb3837